### ISSUE TITLE
DMD Task Item Updates Not Propagating

### ISSUE THREAD

**Issue by nataliemacdonald** [crkn-rcdr/Access-Platform#390]
When: 2021-11-30T19:13:28.623Z

Who: Natalie MacDonald

Affected Slugs: 00169

Maybe it's just in the queue and it's taking more than an hour?

**Comment by RussellMcOrmond** [comment]
I am assuming you are talking about https://www-demo.canadiana.ca/view/oocihm.00169/1?r=0&s=1   ?

It looks to me like the record updated today, soon after you set it to process:

```
  "updateInternalmeta": {
    "requestDate": "2021-11-30T18:00:11Z",
    "processDate": "2021-11-30T18:05:58Z",
    "succeeded": true,
    "message": ""
  },

```

I note that the new record has "Indigenous poeoples -- Canada" rather than the version on the old platform which has "Indians of North America -- Canada".   https://www.canadiana.ca/view/oocihm.00169/1?r=0&s=1



Would it be useful if the developers Developers (@DivyaKrishnan26 , @brittnylapierre ) had https://access.canadiana.ca/object/69429/m0x921c1z21n display something about the success/failure of the process update?  This one passed, but this same interface is likely the best one for displaying any errors.

**Comment by brittnylapierre** [comment]
> I am assuming you are talking about https://www-demo.canadiana.ca/view/oocihm.00169/1?r=0&s=1 ?
> 
> It looks to me like the record updated today, soon after you set it to process:
> 
> ```
>   "updateInternalmeta": {
>     "requestDate": "2021-11-30T18:00:11Z",
>     "processDate": "2021-11-30T18:05:58Z",
>     "succeeded": true,
>     "message": ""
>   },
> ```
> 
> I note that the new record has "Indigenous poeoples -- Canada" rather than the version on the old platform which has "Indians of North America -- Canada". https://www.canadiana.ca/view/oocihm.00169/1?r=0&s=1
> 
> Would it be useful if the developers Developers (@DivyaKrishnan26 , @brittnylapierre ) had https://access.canadiana.ca/object/69429/m0x921c1z21n display something about the success/failure of the process update? This one passed, but this same interface is likely the best one for displaying any errors.

@RussellMcOrmond Could I poll the status of updateInternalmeta in the dmd task page and add it for each item in the table to show a status of when the metadata was updated or not for each item? :o 

We could also display this in the editor if it's helpful like you said. I know this also gets changed when someone publishes/unpublishes something though. 

Or instead of polling for each item in the table, we could link to the item in the editor, then in the editor display the  updateInternalmeta information. 

(fyi @nataliemacdonald )

**Comment by RussellMcOrmond** [comment]
I know what you are trying to do, but it wouldn't accomplish what you want.

`updateInternalmeta` is a field used by Hammer2, which is yet another one of these transitional tools.  Hammer2 successfully finishing its part of the work does *not* mean that something is searchable (that is a few more stages down the metadata bus), nor does it mean that it is visible on the CAP website (also, a different database).

Given we are sometimes talking about anything from 1 to hundreds of thousands of updates in a queue, instant gratification is simply not possible.

Lets please keep focus on the current projects (Prerequisites for adopting Archivematica), and look at this type of optimisation only once other prerequisite decisions have been made (including what software we will be using as the base for different parts of our services.  Hammer2 will be going away soon enough, but is currently not a priority or even work within current projects).

**Comment by RussellMcOrmond** [comment]
These records on the old and new databases look the same at the moment.  I will mark this as closed, but it can be re-opened if needed.

@nataliemacdonald may want to make a specific feature request via the various feedback methods (Here on GitHub, via slack, via email) to get into the feedback that @brittnylapierre and @DivyaKrishnan26 are collecting.
